docs/content/manual/misc/3902-elemental-cloud-init-harvester-crds.md

```diff
@@ -0,0 +1,217 @@
+8. Reboot the node
+9. Validate that the file `/oem/99_ssh_test_single_node.yaml` is not present in the `/oem` directory on that single node
+
+#### Validate that custom labels are reflected upon reboot of nodes with CloudInit Resource
+1. On two or more nodes in your multi-node cluster, create a label on each host of something like `testingmode: testing`
+2. Prepare a YAML loadout of a CloudInit resource that takes the shape of:
+```yaml
+apiVersion: node.harvesterhci.io/v1beta1
+kind: CloudInit
+metadata:
+  name: write-file-reflects-nodes
+spec:
+  matchSelector:
+    testingmode: testing
+  filename: 99_filewrite_nodes.yaml
+  contents: |
+    stages:
+      fs:
+        - name: "write file test reflects nodes"
+          commands:
+            - echo "hello, there" > /etc/sillyfile.conf
+```
+3. validate that the cloudinit resource was built correctly `kubectl describe cloudinits/write-file-reflects-nodes`
+4. Check the nodes applied to, to make sure `/oem/99_filewrite_nodes.yaml` is present, if not you may need to wait for the controller to reconcile the label logic on the nodes
+4. Reboot all nodes that had the label applied
+5. Log into each node and ensure that `sudo cat /etc/sillyfile.conf` exists.
+6. Remove or modify the label on one of the nodes.
+7. Reboot that node.
+7. Validate that the `/oem/99_filewrite_nodes.yaml` file is removed, you may need to wait for a bit for the controller to reconcile something like `watch -n 3 ls -alh /oem` as it may take sometime
+8. Reboot that node again.
+9. Once node is back up validate that `/etc/sillyfile.conf` no longer exists on the node.
+
+#### Validate that CloudInit CRD respects pausing being turned on
+1. On a single node in the cluster have a label you can reference it by, this could be something like `kubernetes.io/hostname`
+2. Prepare a YAML file that takes the shape of something like:
+```yaml
+apiVersion: node.harvesterhci.io/v1beta1
+kind: CloudInit
+metadata:
+  name: silly-file-one-node-no-changes
+spec:
+  matchSelector:
+    kubernetes.io/hostname: harvester-node-0
+  paused: false
+  filename: 99_filewrite_singelnode_wont_get_changes.yaml
+  contents: |
+    stages:
+      fs:
+        - name: "write file test reflects nodes"
+          commands:
+            - echo "hello, there" > /etc/sillyfile-that-doesnt-get-changes.conf
+```
+3. Apply the resource `kubectl create -f`
+4. Reboot the node
+5. Check that the node has the file `cat /etc/sillyfile-that-doesnt-get-changes.conf`
+6. Patch the file with YAML that takes the shape of something like ( `kubectl patch --patch-file your-filepath/filename.yaml cloudinits/silly-file-one-node-no-changes --type=merge` ):
+```yaml
+apiVersion: node.harvesterhci.io/v1beta1
+kind: CloudInit
+metadata:
+  name: silly-file-one-node-no-changes
+spec:
+  matchSelector:
+    kubernetes.io/hostname: harvester-node-0
+  paused: true
+  filename: 99_filewrite_singelnode_wont_get_changes.yaml
```

Comment:
typo, should be `single`?

Reply:
Thanks for this catch :smile: :+1: - updated & force pushed fixing